diff -urN a/configure.in b/configure.in
--- a/configure.in	2012-01-13 17:25:49.000000000 +0100
+++ b/configure.in	2012-03-11 16:50:54.072285628 +0100
@@ -615,8 +615,6 @@
     fi
 
     PKG_CHECK_MODULES(GUI, $gui_modules)
-
-    AC_PATH_PROG([GTK_BUILDER_CONV], gtk-builder-convert)
 elif test "$enable_dbus_service" = "yes"; then
     # syncevo-dbus-server needs localization
     :
diff -urN a/src/gtk-ui/Makefile.in b/src/gtk-ui/Makefile.in
--- a/src/gtk-ui/Makefile.in	2012-01-13 17:29:03.000000000 +0100
+++ b/src/gtk-ui/Makefile.in	2012-03-11 16:52:33.313615324 +0100
@@ -224,7 +224,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
@@ -435,7 +434,7 @@
 
 desktopdir = $(datadir)/applications
 EXTRA_DIST = \
-	ui.glade \
+	$(glade_DATA) \
 	$(themerc_DATA) \
 	$(applications_in_files) \
 	$(noinst_DATA)
diff -urN a/configure b/configure
--- a/configure	2012-03-11 16:55:02.590582074 +0100
+++ b/configure	2012-03-11 17:00:15.834528596 +0100
@@ -857,7 +857,6 @@
 GUI_PROGRAMS
 GETTEXT_PACKAGE
 SYNCEVOLUTION_LOCALEDIR
-GTK_BUILDER_CONV
 GUI_LIBS
 GUI_CFLAGS
 GTK_2_18_LIBS
@@ -18802,45 +18801,6 @@
 	:
 fi
 
-    # Extract the first word of "gtk-builder-convert", so it can be a program name with args.
-set dummy gtk-builder-convert; ac_word=$2
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
-$as_echo_n "checking for $ac_word... " >&6; }
-if test "${ac_cv_path_GTK_BUILDER_CONV+set}" = set; then :
-  $as_echo_n "(cached) " >&6
-else
-  case $GTK_BUILDER_CONV in
-  [\\/]* | ?:[\\/]*)
-  ac_cv_path_GTK_BUILDER_CONV="$GTK_BUILDER_CONV" # Let the user override the test with a path.
-  ;;
-  *)
-  as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-    for ac_exec_ext in '' $ac_executable_extensions; do
-  if { test -f "$as_dir/$ac_word$ac_exec_ext" && $as_test_x "$as_dir/$ac_word$ac_exec_ext"; }; then
-    ac_cv_path_GTK_BUILDER_CONV="$as_dir/$ac_word$ac_exec_ext"
-    $as_echo "$as_me:${as_lineno-$LINENO}: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-  done
-IFS=$as_save_IFS
-
-  ;;
-esac
-fi
-GTK_BUILDER_CONV=$ac_cv_path_GTK_BUILDER_CONV
-if test -n "$GTK_BUILDER_CONV"; then
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: $GTK_BUILDER_CONV" >&5
-$as_echo "$GTK_BUILDER_CONV" >&6; }
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-fi
-
 
 elif test "$enable_dbus_service" = "yes"; then
     # syncevo-dbus-server needs localization
diff -urN a/Makefile.in b/Makefile.in
--- a/Makefile.in	2012-01-13 17:29:03.000000000 +0100
+++ b/Makefile.in	2012-03-11 17:00:43.672657588 +0100
@@ -222,7 +222,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/backends/addressbook/Makefile.in b/src/backends/addressbook/Makefile.in
--- a/src/backends/addressbook/Makefile.in	2012-01-13 17:29:01.000000000 +0100
+++ b/src/backends/addressbook/Makefile.in	2012-03-11 17:04:41.648663300 +0100
@@ -195,7 +195,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/backends/akonadi/Makefile.in b/src/backends/akonadi/Makefile.in
--- a/src/backends/akonadi/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/backends/akonadi/Makefile.in	2012-03-11 17:04:27.081642336 +0100
@@ -192,7 +192,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/backends/buteo/Makefile.in b/src/backends/buteo/Makefile.in
--- a/src/backends/buteo/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/backends/buteo/Makefile.in	2012-03-11 17:05:30.845356838 +0100
@@ -247,7 +247,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/backends/evolution/Makefile.in b/src/backends/evolution/Makefile.in
--- a/src/backends/evolution/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/backends/evolution/Makefile.in	2012-03-11 17:05:43.012539096 +0100
@@ -215,7 +215,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/backends/file/Makefile.in b/src/backends/file/Makefile.in
--- a/src/backends/file/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/backends/file/Makefile.in	2012-03-11 17:05:19.661108521 +0100
@@ -191,7 +191,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/backends/kcalextended/Makefile.in b/src/backends/kcalextended/Makefile.in
--- a/src/backends/kcalextended/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/backends/kcalextended/Makefile.in	2012-03-11 17:04:56.831642866 +0100
@@ -230,7 +230,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/backends/maemo/Makefile.in b/src/backends/maemo/Makefile.in
--- a/src/backends/maemo/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/backends/maemo/Makefile.in	2012-03-11 17:04:16.977321440 +0100
@@ -193,7 +193,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/backends/qtcontacts/Makefile.in b/src/backends/qtcontacts/Makefile.in
--- a/src/backends/qtcontacts/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/backends/qtcontacts/Makefile.in	2012-03-11 17:05:54.099793935 +0100
@@ -228,7 +228,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/backends/sqlite/Makefile.in b/src/backends/sqlite/Makefile.in
--- a/src/backends/sqlite/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/backends/sqlite/Makefile.in	2012-03-11 17:05:07.894899314 +0100
@@ -193,7 +193,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/backends/webdav/Makefile.in b/src/backends/webdav/Makefile.in
--- a/src/backends/webdav/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/backends/webdav/Makefile.in	2012-03-11 17:04:02.283309014 +0100
@@ -194,7 +194,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/backends/xmlrpc/Makefile.in b/src/backends/xmlrpc/Makefile.in
--- a/src/backends/xmlrpc/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/backends/xmlrpc/Makefile.in	2012-03-11 17:06:06.378968667 +0100
@@ -192,7 +192,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/dbus/glib/Makefile.in b/src/dbus/glib/Makefile.in
--- a/src/dbus/glib/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/dbus/glib/Makefile.in	2012-03-11 17:03:05.414131158 +0100
@@ -216,7 +216,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/dbus/interfaces/Makefile.in b/src/dbus/interfaces/Makefile.in
--- a/src/dbus/interfaces/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/dbus/interfaces/Makefile.in	2012-03-11 17:03:28.805559033 +0100
@@ -161,7 +161,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/dbus/Makefile.in b/src/dbus/Makefile.in
--- a/src/dbus/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/dbus/Makefile.in	2012-03-11 17:03:17.013351580 +0100
@@ -173,7 +173,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/dbus/qt/Makefile.in b/src/dbus/qt/Makefile.in
--- a/src/dbus/qt/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/dbus/qt/Makefile.in	2012-03-11 17:03:39.644830532 +0100
@@ -253,7 +253,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/gdbus/Makefile.in b/src/gdbus/Makefile.in
--- a/src/gdbus/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/gdbus/Makefile.in	2012-03-11 17:08:00.355308487 +0100
@@ -201,7 +201,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/gnome-bluetooth/Makefile.in b/src/gnome-bluetooth/Makefile.in
--- a/src/gnome-bluetooth/Makefile.in	2012-01-13 17:29:02.000000000 +0100
+++ b/src/gnome-bluetooth/Makefile.in	2012-03-11 17:07:21.861895566 +0100
@@ -178,7 +178,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/gtk-ui/Makefile.in b/src/gtk-ui/Makefile.in
--- a/src/gtk-ui/Makefile.in	2012-03-11 16:55:02.917560097 +0100
+++ b/src/gtk-ui/Makefile.in	2012-03-11 17:06:47.048235338 +0100
@@ -1159,9 +1159,6 @@
 sync-moblin.desktop: sync.desktop
 	cp $< $@
 
-ui.xml: ui.glade
-	$(GTK_BUILDER_CONV) $< $(@F)
-
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -urN a/src/Makefile.in b/src/Makefile.in
--- a/src/Makefile.in	2012-01-13 17:29:01.000000000 +0100
+++ b/src/Makefile.in	2012-03-11 17:08:49.985972903 +0100
@@ -471,7 +471,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/syncevo/configs/Makefile.in b/src/syncevo/configs/Makefile.in
--- a/src/syncevo/configs/Makefile.in	2012-01-13 17:29:03.000000000 +0100
+++ b/src/syncevo/configs/Makefile.in	2012-03-11 17:07:40.156666004 +0100
@@ -158,7 +158,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/syncevo/Makefile.in b/src/syncevo/Makefile.in
--- a/src/syncevo/Makefile.in	2012-01-13 17:29:03.000000000 +0100
+++ b/src/syncevo/Makefile.in	2012-03-11 17:08:37.272827329 +0100
@@ -314,7 +314,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/src/synthesis-includes/Makefile.in b/src/synthesis-includes/Makefile.in
--- a/src/synthesis-includes/Makefile.in	2012-01-13 17:29:03.000000000 +0100
+++ b/src/synthesis-includes/Makefile.in	2012-03-11 17:07:05.662984270 +0100
@@ -140,7 +140,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
diff -urN a/test/Makefile.in b/test/Makefile.in
--- a/test/Makefile.in	2012-01-13 17:29:03.000000000 +0100
+++ b/test/Makefile.in	2012-03-11 17:08:10.562622472 +0100
@@ -133,7 +133,6 @@
 GTHREAD_LIBS = @GTHREAD_LIBS@
 GTK_2_18_CFLAGS = @GTK_2_18_CFLAGS@
 GTK_2_18_LIBS = @GTK_2_18_LIBS@
-GTK_BUILDER_CONV = @GTK_BUILDER_CONV@
 GUI_CFLAGS = @GUI_CFLAGS@
 GUI_DESKTOP_FILES = @GUI_DESKTOP_FILES@
 GUI_LIBS = @GUI_LIBS@
