--- configure.in	2009-06-21 13:01:46.000000000 +0200
+++ configure.in	2010-10-08 12:24:26.000000000 +0200
@@ -49,12 +49,14 @@ AC_CHECK_LIB(boost_regex, [main], , [
 AC_CHECK_HEADERS(boost/regex.hpp)
 AC_SUBST(DEPS_CFLAGS)
 AC_SUBST(DEPS_LIBS)
+LIBS="$LIBS $DEPS_LIBS"
 
 dnl Special case for poppler-glib < 0.6
 case $($PKG_CONFIG --modversion poppler-glib) in
 	0.5.*)
 		AC_DEFINE(OLD_POPPLER,"1","Using poppler < 0.6");;
 esac
+AC_CHECK_FUNCS([poppler_page_get_selected_text])
 
 
 dnl I18n
--- src/Document.C	2009-08-02 11:16:46.000000000 +0200
+++ src/Document.C	2010-10-08 09:39:22.000000000 +0200
@@ -488,7 +488,11 @@ bool Document::readPDF ()
 			textdump += poppler_page_get_text (page, rect);
 		#else
 			#warning Using poppler >= 0.6
-			textdump += poppler_page_get_text (page, POPPLER_SELECTION_GLYPH, rect);
+			#ifdef HAVE_POPPLER_PAGE_GET_SELECTED_TEXT
+				textdump += poppler_page_get_text (page);
+			#else
+				textdump += poppler_page_get_text (page, POPPLER_SELECTION_GLYPH, rect);
+			#endif
 		#endif
 
 		poppler_rectangle_free (rect);
