--- SConstruct.orig	2012-03-15 23:29:26.655890118 +0100
+++ SConstruct	2012-03-15 23:30:16.166549130 +0100
@@ -79,12 +79,6 @@
     fid.write("clib_path = %s\n" %libpath)
     fid.close()
     print("create clib_path.conf file (libfile: %s)"%(libpath))
-    #run python installer
-    if "win" in os.sys.platform or env['DESTDIR'] == "" or env['DESTDIR'] == "${DESTDIR}":
-        python_installer = subprocess.Popen("python setup.py install --prefix=%s"%(env['PREFIX']),shell=True)
-    else:
-        python_installer = subprocess.Popen("python setup.py install --root=%s --prefix=%s" %(env['DESTDIR'],env['PREFIX']),shell=True)
-    python_installer.wait()
 
 ############################
 #    config like things
