--- SConstruct.orig	2012-03-21 16:56:45.496562596 +0100
+++ SConstruct	2012-03-21 16:57:03.643331454 +0100
@@ -81,13 +81,6 @@
     fid.close()
     print("create clib_path.conf file (libfile: %s)"%(libpath))
     #run python installer
-    if platform.linux_distribution(full_distribution_name=0)[0] in ["Ubuntu","debian"]:
-        python_installer = subprocess.Popen("python setup.py install --prefix=%s --install-layout=deb" %(env['PREFIX']),shell=True)
-    elif "win" in os.sys.platform or env['DESTDIR'] == "" or env['DESTDIR'] == "${DESTDIR}":
-        python_installer = subprocess.Popen("python setup.py install --prefix=%s"%(env['PREFIX']),shell=True)
-    else:
-        python_installer = subprocess.Popen("python setup.py install --root=%s --prefix=%s" %(env['DESTDIR'],env['PREFIX']),shell=True)
-    python_installer.wait()
 
 ############################
 #    config like things
