--- SConstruct.orig	2012-03-02 16:55:48.553161142 +0100
+++ SConstruct	2012-03-02 16:56:58.713438388 +0100
@@ -67,12 +67,6 @@
     fid.write("clib_path = %s\n" %libpath)
     fid.close()
     print("create clib_path.conf file (libfile: %s)"%(libpath))
-    #run python installer
-    if "win" in os.sys.platform:
-        python_installer = subprocess.Popen("python setup.py install --prefix="+env['PREFIX'],shell=True)
-    else:
-        python_installer = subprocess.Popen("python setup.py install --prefix="+env['DESTDIRPREFIX'],shell=True)
-    python_installer.wait()
 
 ############################
 #    config like things
